javalib/src/main/scala/java/util/concurrent/ConcurrentHashMap.scala

```diff
@@ -0,0 +1,138 @@
+package java.util.concurrent
+
+import scala.collection.JavaConversions.{ setAsJavaSet, asScalaSet, asJavaCollection, asJavaEnumeration }
+
+trait ConcurrentMap[K, V] extends java.util.Map[K, V] {
+
+  def putIfAbsent(key: K, value: V): V =
+    if (!containsKey(key))
+      put(key, value)
+    else
+      get(key)
+
+  def remove(key: Any, value: Any): Boolean =
+    if (containsKey(key) && get(key).equals(value)) {
+      remove(key)
+      true
+    } else false
+
+  def replace(key: K, value: V): V =
+    if (containsKey(key)) {
+      put(key, value)
+    } else get(key)
+
+  def replace(key: K, oldValue: V, newValue: V): Boolean =
+    if (containsKey(key) && get(key).equals(oldValue)) {
+      put(key, newValue)
+      true
+    } else false
+
+}
+
+class ConcurrentHashMap[K >: Null, V >: Null](initialCapacity: Int = 16,
+  loadFactor: Float = 0.75F,
+  concurrencyLevel: Int = 16)
```

Comment:
Also, never use default parameters when implementing a Java signature. It does not work (not binary compatible).